src/ValueParsers/MonthNameProvider.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace ValueParsers;
+
+/**
+ * A common interface for both localizing month names in formatters as well as unlocalizing month
+ * names in parsers.
+ *
+ * @since 0.8.3
+ *
+ * @licence GNU GPL v2+
+ * @author Thiemo Mättig
+ */
+interface MonthNameProvider {
+
+	/**
+	 * @param string $languageCode
+	 *
+	 * @return string[] Array mapping month numbers (1 to 12) to localized month names.
+	 */
+	public function getLocalizedMonthNames( $languageCode );
+
+	/**
+	 * @param string $languageCode
+	 *
+	 * @return int[] Array mapping localized month names (possibly including full month names,
+	 * genitive names and abbreviations) to month numbers (1 to 12).
+	 */
+	public function getMonthNumbers( $languageCode );
+
+}
```

Comment:
1. See TimeParserFactory::getMonthNameUnlocalizer. It needs month names in two languages.
2. See basically all date parsers that support months. They all should support at least abbreviated and non-abbreviated month names.